apps/events/viewsets.py

```diff
@@ -198,17 +217,38 @@ def delete_from_favorite(self, request, event_id: int):
     )
     @action(
         methods=['get'],
-        detail=False,
-        permission_classes=[IsAuthenticated],
+        detail=True,
+        permission_classes=[IsAuthenticated, EventPermissions],
         url_path='private-url',
         url_name='private_url',
         filter_backends=[],
         pagination_class=None
     )
-    def set_private_url(self, request):
-        private_url = str(uuid4())
+    def get_private_url(self, request, event_id: int):
+        event = self.get_object()
+        private_url = event.private_url
         return Response(private_url, status=status.HTTP_200_OK)
 
+    @swagger_auto_schema(
+        request_body=no_body
+    )
+    @action(
+        methods=['get'],
+        detail=False,
+        permission_classes=[IsAuthenticated],
+        url_path='private/<str:token>',
+        url_name='private',
+    )
+    def get_private_event(self, request, token: str):
+        Event.objects.filter(private_url=token).retrieve(request)
+
+    @swagger_auto_schema(
+        request_body=no_body,
+    )
+    @get_private_event.mapping.delete
+    def delete_private_event(self, request, token: str):
+        Event.objects.filter(private_url=token).destroy(request)
```

Comment:
Deletion should be probably removed due to the fact that only an owner can delete an event and the owner can get access without token using private link. So the owner will just use basic delete call with OwnerPermissios

Reply:
Done:
    def get_private_event(self, request, token):
        event_instance = self.get_object()
        serializer = EventRetrieveSerializer(event_instance)
        return Response(serializer.data)